doc/source/user/build_system/core_files.rst

```diff
@@ -0,0 +1,257 @@
+  For example, a list of items can be written in the following two, semantically identical ways.
+
+  .. code-block:: yaml
+
+     [ item1, item2 ]
+
+  is semantically identical to
+
+  .. code-block:: yaml
+
+     - item1
+     - item2
+
+  The same is true for dictionaries (key/value pairs).
+
+  .. code-block:: yaml
+
+     { key1: value1, key2: value2 }
+
+  is semantically identical to
+
+  .. code-block:: yaml
+
+     key1: value1
+     key2: value2
+
+  In most cases, the longer (second) form is preferred, as it is easier to make changes while keeping the diff easy to read.
+
+For a quick introduction into most of YAML's features have a look at `Learn YAML in Y minutes <https://learnxinyminutes.com/docs/yaml/>`_.
+The full YAML 1.2 specification is available at `yaml.org <https://yaml.org/spec/1.2/spec.html>`_ (it's not an easy read, though).
+
+An example: the blinky core
+---------------------------
+
+The following sections explain how to add FuseSoC support to a hardware project.
+The code is taken from an example design in the `FuseSoC source tree <https://github.com/olofk/fusesoc>`_ in the ``tests/userguide/blinky``.
+
+The design consists of two SystemVerilog files, a testbench, a Xilinx constraint file (with pin mappings for a Nexys Video FPGA board), and finally, the FuseSoC core file.
+
+.. code-block:: console
+
+   $ tree tests/userguide/blinky/
+   tests/userguide/blinky/
+   ├── blinky.core
+   ├── data
+   │   └── nexys_video.xdc
+   ├── rtl
+   │   ├── blinky.sv
+   │   └── macros.svh
+   └── tb
+      └── blinky_tb.sv
+
+   3 directories, 5 files
+
+To get started, here's the full ``blinky.core`` file.
+The following sections will refer back to this example to discuss it in detail.
+
+.. literalinclude:: ../../../../tests/userguide/blinky/blinky.core
+   :language: yaml
+   :caption: ``blinky.core``, an exemplary core file
+   :name: ug_build_system_cb_blinky_full
+   :linenos:
+
+Naming the core file
+--------------------
+
+The core file can have any name, but it must end in ``.core``.
+It is recommended to choose a file name matching the core name, as discussed below.
+
+The first line: ``CAPI=2``
+--------------------------
+
+A core file always starts with the line ``CAPI=2``.
+No other content (including comments) is allowed before this line, as FuseSoC uses this line to differentiate between different versions of the CAPI schema.
+Only CAPI version 2 is specified at the moment.
+
+.. _ug_build_system_core_name:
+
+The core name, version, and description
+---------------------------------------
+
+Each core has a name, given in the ``name`` key.
+Core names can be freely chosen, but need to follow a common structure called *VLNV*.
+:term:`VLNV` stands the four parts of a core name, which are separated by colon (``:``): Vendor, Library, Name, and Version.
+
+.. todo::
+
+   Specify version number requirements, and what happens if no version number is given.
```

Comment:
Maybe add a note that semver is highly recommended (might require that at some point in the distant future)